src/platform/plugins/shared/discover/public/application/main/state_management/redux/internal_state.ts

```diff
@@ -139,8 +114,19 @@ export const internalStateSlice = createSlice({
         {}
       );
       state.tabs.allIds = action.payload.allTabs.map((tab) => tab.id);
+      state.tabs.recentlyClosedTabsById = action.payload.recentlyClosedTabs.reduce<
+        Record<string, RecentlyClosedTabState>
+      >(
+        (acc, tab) => ({
+          ...acc,
+          [tab.id]: tab,
+        }),
+        {}
+      );
+      state.tabs.recentlyClosedTabIds = action.payload.recentlyClosedTabs.map((tab) => tab.id);
```

Comment:
After playing around with the changes, I understand more what was being done here, but I think we can achieve the same thing without duplicating the tabs state and keeping things relational. Instead of having a single `byId` object, we can split the recently closed tabs into a separate `recentlyClosedTabsById` object. That way it supports both open and closed tabs with the same ID, and we can resolve the `selectTab` case directly in the selector.